proxy/apifilterproxy/handler.go

```diff
@@ -0,0 +1,182 @@
+		//build router paths
+		for _, path := range filter.Paths {
+			for _, method := range filter.Methods {
+				log.Debugf("Adding route: %v %v", strings.ToUpper(method), path)
+				router.Methods(strings.ToUpper(method)).Path(path).HandlerFunc(http.HandlerFunc(handleRequest))
+			}
+		}
+	}
+	router.Methods("POST").Path("/v1-api-filter-proxy/reload").HandlerFunc(http.HandlerFunc(reload))
+	router.NotFoundHandler = http.HandlerFunc(handleNotFoundRequest)
+
+	return router
+}
+
+func handleRequest(w http.ResponseWriter, r *http.Request) {
+	path, _ := mux.CurrentRoute(r).GetPathTemplate()
+
+	log.Debugf("Request Path matched: %v", path)
+
+	bodyBytes, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Errorf("Error reading request Body %v for path %v", r, path)
+		returnHTTPError(w, r, http.StatusBadRequest, fmt.Sprintf("Error reading json request body, err: %v", err))
+		return
+	}
+
+	var jsonInput map[string]interface{}
+	if len(bodyBytes) > 0 {
+		err = json.Unmarshal(bodyBytes, &jsonInput)
+		if err != nil {
+			log.Errorf("Error unmarshalling json request body: %v", err)
+			returnHTTPError(w, r, http.StatusBadRequest, fmt.Sprintf("Error reading json request body: %v", err))
+			return
+		}
+	}
+
+	headerMap := make(map[string][]string)
+	for key, value := range r.Header {
+		headerMap[key] = value
+	}
+
+	api := r.URL.Path
+
+	inputBody, inputHeaders, destination, proxyErr := manager.ProcessPreFilters(path, api, jsonInput, headerMap)
+	if proxyErr.Status != "" {
+		//error from some filter
+		log.Debugf("Error from proxy filter %v", proxyErr)
+		writeError(w, proxyErr)
+		return
+	}
+
+	jsonStr, err := json.Marshal(inputBody)
+	r.Body = ioutil.NopCloser(bytes.NewReader(jsonStr))
+	r.ContentLength = int64(len(jsonStr))
+
+	for key, value := range inputHeaders {
+		for _, singleVal := range value {
+			r.Header.Add(key, singleVal)
+		}
+	}
+
+	destProxy, err := newProxy(destination)
+	if err != nil {
+		log.Errorf("Error creating a reverse proxy for destination %v", destination)
+		returnHTTPError(w, r, http.StatusInternalServerError, fmt.Sprintf("Error creating a reverse proxy for destination %v", destination))
+		return
+	}
+	destProxy.reverseProxy.ServeHTTP(w, r)
+}
+
+func handleNotFoundRequest(w http.ResponseWriter, r *http.Request) {
+	destProxy, err := newProxy(manager.DefaultDestination)
+	if err != nil {
+		log.Errorf("Error creating a reverse proxy for destination %v", manager.DefaultDestination)
+		returnHTTPError(w, r, http.StatusInternalServerError, fmt.Sprintf("Error creating a reverse proxy for destination %v", manager.DefaultDestination))
+		return
+	}
+	destProxy.reverseProxy.ServeHTTP(w, r)
+}
+
+func reload(w http.ResponseWriter, r *http.Request) {
+	log.Info("Reload proxy config")
+	err := manager.Reload()
+	if err != nil {
+		//failed to reload the config from the config.json
+		log.Debugf("Reload failed with error %v", err)
+		returnHTTPError(w, r, http.StatusInternalServerError, "Failed to reload the proxy config")
```

Comment:
shall we report the exact error here? Given it is getting logged in Debug, user wont have content of what has happened